### DISCUSSION TITLE
The "fetch-depth" value purpose

### DISCUSSION THREAD

**Discussion by aleks-ivanov** [testspace-com/support#5] [Q&A]
Hi!

I was wondering, what is the purpose of setting exactly 50 to the checkout fetch-depth ?

Would lower or higher value be detrimental in any way ?

**Comment by munderseth** [comment]
Hi @AleksIvanovSinglet, 

We use the `fetch-depth` to collect the commits used for the **Code Change** metric suite and associated graph within the *Metrics* tab. Most other CIs default the `fetch-depth` to 50 (i.e., Circle CI, Azure, Travis, etc.). GitHub decided to default their depth to 1.  We are not sure exactly why GitHub defaulted to 1. It is our belief there is no real performance difference with different fetch depths. 

Note that we are making changes in a future release to collect the **Code Change** independent of the setting.
